### DISCUSSION TITLE
My own video client

### DISCUSSION THREAD

**Discussion by roycw** [bluenviron/mediamtx#3784] [Questions]
### Question

Would love some instruction on how i can consume the feeds in my own raw video element in the browser.

**Comment by Taps7734** [comment]
There isn't much info here, but I am going to assume you would like to view a web page that shows your video.

If you have WebRTC enabled you can open the URL something like 

https://my.server.com:8889/Path/To/Stream

"HTTP(S)" will depend on if you have set up certificates for encryption.
"my.server.com" of course is the FQDN for your server.
"8889" is the default port for WebRTC.
"/Path/To/Stream" is what you have set up in the config file or in your broadcasting software.


To view an RTMP stream, you would have to use a client like VLC or something using a similar URL like rtmp://my.server.com:1935/Path/To/Stream

By default, correct me if I am wrong, without a browser plugin, you cannot view RTMP streams directly in a browser.

**Reply by aler9** [reply]
this is not related to the question, that was related to embedding, and mostly wrong.

**Comment by aler9** [comment]
Hello, raw video tags do not support real-time video streams by themselves, they need to be initialized through JavaScript. Since this is too complex for most users, the README suggests two approaches:

* embedding the web page provided by the server through an iframe, as described here: https://github.com/bluenviron/mediamtx?tab=readme-ov-file#web-browsers-1
* building a custom page upon the source code of the web page provided by the server, and hosting it somewhere else, as described here: https://github.com/bluenviron/mediamtx?tab=readme-ov-file#web-browsers-1

[a comment by an automated account is not shown]
